File: scripts/benchmarks/basic.jl

n = 10000;
a = rand(Float32,(n,n))
b = rand(Float32,(n,n))
c = zeros(Float32,(n,n));
t1 = time();

for i = 1:n
  for j = 1:n
    c[i,j] = a[i,j] + b[i,j]
  end
end

t2 = time();

for i = 1:10
c = a+ b;	
end

t3 = time();

dt1 = t2 - t1
dt2 = t3 - t2
n2 = n*n

mflops1 = n2 / dt1 / 1e6;
mflops2 = n2 / dt2 / 1e5;

println("times $dt1,$dt2, mflops $mflops1,$mflops2")

File: scripts/benchmarks/basic.py
import time
import numpy as np;
import numpy.random as rand;
t0 = time.time()
n = 10000
a = rand.rand(n,n)
b = rand.rand(n,n)

t0 = time.time()
c = a + b;
t1 = time.time()
dt1 = t1 - t0
print dt1

for i in range(0,n):
    for j in range(0,n):
        c[i][j] = a[i][j] + b[i][j];

t2 = time.time()

dt2 = t2 - t1
print dt1, dt2
